obsplus/utils.py

```diff
@@ -989,3 +1032,151 @@ def _column_contains(ser: pd.Series, str_sequence: Iterable[str]) -> pd.Series:
     """ Test if a str series contains any values in a sequence """
     safe_matches = {re.escape(x) for x in str_sequence}
     return ser.str.contains("|".join(safe_matches)).values
+
+
+# --- functions for dealing with time conversions (ugh)
+
+
+@singledispatch
+def to_datetime64(value: utc_able_type, default=pd.NaT) -> np.datetime64:
+    """
+    Convert time value to a numpy datetime64.
+
+    Parameters
+    ----------
+    value
+        Any Value that can be interpreted as a time.
+    default
+        A value to return if value is null. Default
+    """
+    if pd.isnull(value) or not value:
+        if not pd.isnull(default):
+            return to_datetime64(default)
+        return default
+    elif isinstance(value, np.datetime64):
+        return value
+    elif isinstance(value, pd.Timestamp):
+        return value.to_numpy()
+    try:
+        utc = obspy.UTCDateTime(value)
+        return np.datetime64(utc._ns, "ns")
+    # the UTCDateTime is too big or small, convert
+    except (SystemError, OverflowError):
+        new = LARGEDT64 if np.sign(utc._ns) > 0 else SMALLDT64
+        msg = (
+            f"{utc} is too large to represent with a int64 with ns precision,"
```

Comment:
Well since we will both be dead I guess we let someone else worry about it? This was one of the drawbacks discussed in #88. Can you think of a better way to handle it? Maybe it should raise an exception rather than warn to change the value?